### PR TITLE
Make s3acl take an optional value

### PR DESCRIPTION
**Pull request by hkupty**
So we can resort back to inheriting the bucket default setting instead of
explicitly setting a value for it.

This depends on ohnosequences/ivy-s3-resolver#20.

I'll fix the comments there and then update here. Still I wanted to open this so the
desing of this change can be debated as well.

Also, ref #54.

### CODE REVIEW

File: src/main/scala/SBTS3Resolver.scala [hunk 1]
```diff
@@ -85,7 +86,9 @@ object SbtS3Resolver extends AutoPlugin {
     lazy val s3region       = settingKey[Region]("AWS Region for your S3 resolvers")
     lazy val s3overwrite    = settingKey[Boolean]("Controls whether publishing resolver can overwrite artifacts")
     lazy val s3sse          = settingKey[Boolean]("Controls whether publishing resolver will use server side encryption")
+    @deprecated("Use s3optAcl instead. This currently forwards the value to s3optAcl, but is overwritten by it.")
     lazy val s3acl          = settingKey[S3ACL]("Controls whether published artifacts are accessible publicly via http(s) or not")
+    lazy val s3optAcl       = settingKey[Option[S3ACL]]("Controls published artifacts visibility via http(s) or inherits bucket setting")
```
Comment (laughedelic) [inline src/main/scala/SBTS3Resolver.scala]: I suggest a bit different approach: keep one key `s3acl`, change its type to `Option[S3ACL]` and add an implicit conversion `S3ACL => Some[S3ACL]`. Then you can _deprecate that implicit conversion_ with an explanation that the type has changed and a suggestion to write `Some(...)` explicitly.

This way things also don't break (from users' point of view) and plus we keep the same setting key.

Comment (hkupty) [reply src/main/scala/SBTS3Resolver.scala]: Makes sense. :+1:

**Comment by hkupty** [comment]
Seems to be complete now. I'm leaving the PR open for you to edit, just so you can bump the ivy-s3-resolver version.

Thanks for helping and accepting the PR.
Cheers,
Henry Kupty

**Comment by laughedelic** [comment]
Done! 🎉 Released in [v0.18.0](https://github.com/ohnosequences/sbt-s3-resolver/releases/tag/v0.18.0).

Thanks for the contribution @hkupty 👍

**Pull request merged**